tutorials/Python/11_OOP/OOP.ipynb

```diff
@@ -0,0 +1,769 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the following sections we will have a first look at the simple concepts of object-oriented programming (OOP) to get an idea of this powerful paradigm."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Molecule Class"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A _class_ is the definition of an object. We can think of a class as an architectural plan and an object (usually called an _instance_ of the class in this context) as an house built following said plan. With the same plan, you can build multiple houses (create multiple instances), each housing different people (different data). For our example above, a yet to be defined `Molecule` class would represent a generic molecule (by defining the data and the functionality associated to a molecule) while `acetylene` would be an object (instance) of said class, containing the actual data defining `acetylene`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Creating a new class in Python has a similar syntax to a function definition, with the `class` keyword instead of `def` and without `()` (`()` is optional for a class definition):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class EmptyClass:\n",
+    "    \"\"\"\n",
+    "    Empty class that does nothing.\n",
+    "\n",
+    "    Note:\n",
+    "        The `pass` statement in Python does nothing.\n",
+    "    \"\"\"\n",
+    "    pass"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "_Note #1_: the `pass` statement in Python does nothing and it is therefore useful when a statement is required but no code needs to be written."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "_Note #2_: docstrings are used to document classes as well."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once our class is defined, we can create (_instantiate_) an object of said class:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ec = EmptyClass()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can check that the content of the variable `ec` is indeed an instance of  the class `EmptyClass` using the `isinstance` function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": [
+       "True"
```

Comment:
I can't remember what we agreed on last year, should we be clearing the notebooks of the solutions so that they can execute the cells themselves (I _think_ that's what I'm seeing from the other notebooks)?